capabilityquerier/cmd/main.go

```diff
@@ -25,100 +23,131 @@ func failOnError(err error) {
+	qw, ok := (*args)["qw"].(*capabilityquerier.QueueWorkers)
+	if !ok {
+		return fmt.Errorf("unable to cast capabilityquerier QueueWorkers from arguments")
+	}
+	ctx, ok := (*args)["ctx"].(context.Context)
+	if !ok {
+		return fmt.Errorf("unable to cast context from arguments")
+	}
+	numWorkers, ok := (*args)["numWorkers"].(int)
+	if !ok {
+		return fmt.Errorf("unable to cast numWorkers to int from arguments")
+	}
+	errs, ok := (*args)["errs"].(chan error)
+	if !ok {
+		return fmt.Errorf("unable to cast errs to chan error from arguments")
+	}
+	client, ok := (*args)["client"].(*http.Client)
+	if !ok {
+		return fmt.Errorf("unable to cast client to http.Client from arguments")
+	}
+	jobDuration, ok := (*args)["jobDuration"].(time.Duration)
+	if !ok {
+		return fmt.Errorf("unable to cast jobDuration to time.Duration from arguments")
+	}
+	mq, ok := (*args)["mq"].(*lanternmq.MessageQueue)
+	if !ok {
+		return fmt.Errorf("unable to cast mq to MessageQueue from arguments")
+	}
+	ch, ok := (*args)["ch"].(*lanternmq.ChannelID)
+	if !ok {
+		return fmt.Errorf("unable to cast ch to ChannelID from arguments")
+	}
+	qName, ok := (*args)["qName"].(string)
+	if !ok {
+		return fmt.Errorf("unable to cast qName to string from arguments")
+	}
+
+	// Handle the start message that is sent before the endpoints and the stop message that is sent at the end
+	if string(message) == "start" {
+		err := qw.Start(ctx, numWorkers, errs)
+		failOnError(err)
+		return nil
+	}
+	if string(message) == "stop" {
+		log.Info("Stopping queue workers")
+		err := qw.Stop()
+		failOnError(err)
+		return nil
+	}
 
-	for i, endpointEntry := range listOfEndpoints.Entries {
-		if i%10 == 0 {
-			log.Infof("Processed %d/%d messages", i, len(listOfEndpoints.Entries))
+	urlString := string(message)
+	// Specifically query the FHIR endpoint metadata
+	metadataURL, err := url.Parse(urlString)
+	if err != nil {
+		log.Warn("endpoint URL parsing error: ", err.Error())
+	} else {
+		metadataURL.Path = path.Join(metadataURL.Path, "metadata")
+
+		job := capabilityquerier.Job{
+			Context:      ctx,
+			Duration:     jobDuration,
+			FHIRURL:      metadataURL,
+			Client:       client,
+			MessageQueue: mq,
+			Channel:      ch,
+			QueueName:    qName,
 		}
-		var urlString = endpointEntry.FHIRPatientFacingURI
-		// Specifically query the FHIR endpoint metadata
-		metadataURL, err := url.Parse(urlString)
+
+		err = qw.Add(&job)
 		if err != nil {
-			log.Warn("endpoint URL parsing error: ", err.Error())
-		} else {
-			metadataURL.Path = path.Join(metadataURL.Path, "metadata")
-
-			job := capabilityquerier.Job{
-				Context:      ctx,
-				Duration:     jobDuration,
-				FHIRURL:      metadataURL,
-				Client:       client,
-				MessageQueue: mq,
-				Channel:      ch,
-				QueueName:    qName,
-			}
-
-			err = qw.Add(&job)
-			if err != nil {
-				log.Warn("error adding job to queue workers: ", err.Error())
-				break
-			}
+			log.Warn("error adding job to queue workers: ", err.Error())
```

Comment:
should this go into the `errs` error channel? where does this print to as a go routine?

Reply:
it prints to the console as well, but you're right it would make more sense to have it go to the error channel instead